torch/testing/_internal/common_methods_invocations.py

```diff
@@ -427,6 +427,16 @@ def sample_inputs(self, device, dtype, requires_grad=False):
                                 device_type='cpu', dtypes=[torch.cfloat, torch.cdouble],
                                 active_if=(IS_MACOS or IS_WINDOWS)),
                    )),
+    UnaryUfuncInfo('exp',
```

Comment:
If exp is added here it should be removable from somewhere else, right?

Reply:
Found this one in `test_torch`,
https://github.com/pytorch/pytorch/blob/ec256ab2f2f7c65c3fa25e16438a467f9959937c/test/test_torch.py#L10881-L10939

With changes in this PR, the skipped tests in the test above pass while it fails on master (both tested locally)

Will update the PR to activate the skipped tests.

<details>
<summary>Stack Trace of failed case on master</summary>
```
=================================================================================== FAILURES ===================================================================================
________________________________________________________________ TestTorchDeviceTypeCPU.test_exp_cpu_complex128 ________________________________________________________________

self = <test_torch.TestTorchDeviceTypeCPU testMethod=test_exp_cpu_complex128>, device = 'cpu', dtype = torch.complex128

    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
    @dtypes(*(torch.testing.get_all_fp_dtypes(include_half=False) +
              torch.testing.get_all_complex_dtypes()))
    @dtypesIfCUDA(*(torch.testing.get_all_fp_dtypes(include_half=True) +
                    torch.testing.get_all_complex_dtypes()))
    def test_exp(self, device, dtype):
        for v in (2, -2) + ((1j, 1 + 1j) if dtype.is_complex else ()):
            a = torch.tensor(v, dtype=dtype, device=device) * torch.arange(18, device=device) / 3 * math.pi
            a = a.to(dtype)
            if dtype == torch.bfloat16:
                with self.assertRaises(TypeError):  # compare_with_numpy doesn't support bfloat16
                    self.compare_with_numpy(torch.exp, np.exp, a)
                return
            self.compare_with_numpy(torch.exp, np.exp, a)
    
            if dtype.is_complex:
                inf_real_zero_imag_in = torch.tensor(complex(float('inf'), 0), device=device, dtype=dtype)
                inf_real_zero_imag_out = torch.exp(inf_real_zero_imag_in).item()
                self.assertTrue(math.isinf(inf_real_zero_imag_out.real))
>               self.assertEqual(inf_real_zero_imag_out.imag, 0, atol=0, rtol=0)

test/test_torch.py:10900: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../.conda/envs/PytorchENV/lib/python3.7/site-packages/torch/testing/_internal/common_utils.py:1149: in assertEqual
    self.assertTrue(result, msg=msg)
E   AssertionError: False is not true : Scalars failed to compare as equal! Comparing nan and 0 gives a difference of nan, but the allowed difference with rtol=0 and atol=0 is only 0!
________________________________________________________________ TestTorchDeviceTypeCPU.test_exp_cpu_complex64 _________________________________________________________________

self = <test_torch.TestTorchDeviceTypeCPU testMethod=test_exp_cpu_complex64>, device = 'cpu', dtype = torch.complex64

    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
    @dtypes(*(torch.testing.get_all_fp_dtypes(include_half=False) +
              torch.testing.get_all_complex_dtypes()))
    @dtypesIfCUDA(*(torch.testing.get_all_fp_dtypes(include_half=True) +
                    torch.testing.get_all_complex_dtypes()))
    def test_exp(self, device, dtype):
        for v in (2, -2) + ((1j, 1 + 1j) if dtype.is_complex else ()):
            a = torch.tensor(v, dtype=dtype, device=device) * torch.arange(18, device=device) / 3 * math.pi
            a = a.to(dtype)
            if dtype == torch.bfloat16:
                with self.assertRaises(TypeError):  # compare_with_numpy doesn't support bfloat16
                    self.compare_with_numpy(torch.exp, np.exp, a)
                return
            self.compare_with_numpy(torch.exp, np.exp, a)
    
            if dtype.is_complex:
                inf_real_zero_imag_in = torch.tensor(complex(float('inf'), 0), device=device, dtype=dtype)
                inf_real_zero_imag_out = torch.exp(inf_real_zero_imag_in).item()
                self.assertTrue(math.isinf(inf_real_zero_imag_out.real))
>               self.assertEqual(inf_real_zero_imag_out.imag, 0, atol=0, rtol=0)

test/test_torch.py:10900: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../.conda/envs/PytorchENV/lib/python3.7/site-packages/torch/testing/_internal/common_utils.py:1149: in assertEqual
    self.assertTrue(result, msg=msg)
E   AssertionError: False is not true : Scalars failed to compare as equal! Comparing nan and 0 gives a difference of nan, but the allowed difference with rtol=0 and atol=0 is only 0!
=============================================================================== warnings summary ===============================================================================
../../.conda/envs/PytorchENV/lib/python3.7/site-packages/torch/nn/modules/conv.py:29
  /home/kshiteej/.conda/envs/PytorchENV/lib/python3.7/site-packages/torch/nn/modules/conv.py:29: DeprecationWarning: invalid escape sequence \_
    :math:`\\frac{\\text{out\_channels}}{\\text{in\_channels}}`).""",  # noqa: W605

test/test_torch.py::TestTorchDeviceTypeCUDA::test_exp_cuda_complex128
test/test_torch.py::TestTorchDeviceTypeCUDA::test_exp_cuda_complex64
  /home/kshiteej/.conda/envs/PytorchENV/lib/python3.7/site-packages/torch/testing/_internal/common_utils.py:901: RuntimeWarning: invalid value encountered in exp
    np_result = np_fn(a)

test/test_torch.py::TestTorchDeviceTypeCUDA::test_exp_cuda_float16
  /home/kshiteej/.conda/envs/PytorchENV/lib/python3.7/site-packages/torch/testing/_internal/common_utils.py:901: RuntimeWarning: overflow encountered in exp
    np_result = np_fn(a)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================================================================== short test summary info ============================================================================
FAILED test/test_torch.py::TestTorchDeviceTypeCPU::test_exp_cpu_complex128 - AssertionError: False is not true : Scalars failed to compare as equal! Comparing nan and 0 give...
FAILED test/test_torch.py::TestTorchDeviceTypeCPU::test_exp_cpu_complex64 - AssertionError: False is not true : Scalars failed to compare as equal! Comparing nan and 0 gives...
==================================================== 2 failed, 27 passed, 2 skipped, 8255 deselected, 4 warnings in 14.80s =====================================================
```
</details>